jupyter_server/serverapp.py

```diff
@@ -1355,9 +1356,7 @@ def _default_root_dir(self):
         else:
             return os.getcwd()
 
-    @validate('root_dir')
-    def _root_dir_validate(self, proposal):
-        value = proposal['value']
+    def _normalize_root_or_preferred_dir(self, value):
```

Comment:
Could we make this name more generic since it doesn't require/act on `root_dir` or `preferred_dir`?
```suggestion
    def _normalize_dir(self, value):
```

Reply:
Fixed!